tools/format/CMakeLists.txt

```diff
@@ -4,76 +4,67 @@
 cmake_minimum_required(VERSION 3.29.0)
 project(msvc_standard_libraries_format NONE)
 
-set(did_search OFF)
-
-if(NOT DEFINED CLANG_FORMAT)
-    message(STATUS "Searching for VS clang-format")
-    set(did_search ON)
-endif()
-
-if(PROJECT_IS_TOP_LEVEL)
-    set(message_level FATAL_ERROR)
-else()
-    set(message_level WARNING)
-endif()
-
 find_program(CLANG_FORMAT
     NAMES clang-format
-    PATHS "C:/Program Files/Microsoft Visual Studio/2022/Preview/VC/Tools/Llvm/bin"
     DOC "The clang-format program to use"
-    NO_DEFAULT_PATH
-    NO_CMAKE_PATH
-    NO_CMAKE_ENVIRONMENT_PATH
-    NO_SYSTEM_ENVIRONMENT_PATH
-    NO_CMAKE_SYSTEM_PATH
+    REQUIRED
 )
 
-if(CLANG_FORMAT)
-    if(did_search)
-        message(STATUS "Searching for VS clang-format - found")
-    endif()
+execute_process(
+    COMMAND "${CLANG_FORMAT}" --version
+    OUTPUT_VARIABLE clang_format_version
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+)
 
-    file(GLOB_RECURSE maybe_clang_format_files
-        "../../benchmarks/inc/*"
-        "../../benchmarks/src/*"
-        "../../stl/inc/*"
-        "../../stl/modules/*"
-        "../../stl/src/*"
-        "../../tests/*"
-        "../../tools/*"
-    )
-    set(clang_format_files "")
+if(clang_format_version MATCHES "clang-format version ([0-9]+\.[0-9]+\.[0-9]+)")
+    set(expected_version "17.0.3")

```

Comment:
Is having a the same patch version `.3` really needed? I think even VS updates can often change that so you will have to update that value pretty often.

Reply:
Good question - it is very strict, but VS doesn't actually update Clang/LLVM very often; we tend to capture exactly one release of a major version, and not even always that frequently (e.g. it's sure looking like we're going to skip Clang 18).

If this turns out to be a problem, we can reduce the strictness in the future. I'd like to start off with strict validation, since in theory every patch version could change formatting behavior.

Reply:
It might not be direct problem for STL, but usually contributors tend to work on other projects too and they might have lets say a newer version of 17.0.x, because of some kind of issue it would be a bit inconvenient to always change `PATH` to satisfy projects.

Reply:
I pushed a one-line change to prefer the clang-format from the VS installation if it's present. Even with another clang-format earlier on `PATH`, this will use the required version without requiring user intervention.